app/views/layouts/_shopfront_script.html.haml

```diff
@@ -0,0 +1,29 @@
+:javascript
+  $(document).ready(function() {
+    var in_iframe = function(){
+      try {
+        return window.self !== window.top;
+      } catch (e) {
+        return true;
+      }
+    };
+
+    var embedded_styles_active = $('body.off-canvas').hasClass('embedded');
+
+    var set_shopfront_styles = function(state) {
+      $.ajax({
+        url: '/embedded_shopfront/'+state,
+        type: 'POST'
+      });
+    };
+
+    if (in_iframe() && !embedded_styles_active){
+      $('body.off-canvas').addClass('embedded');
```

Comment:
The only thing that stops us is `response.header` but generaly I could call `$('body.off-canvas').addClass('embedded')` from `console` or `window.top `and I would have embedded shopfront which looks vulnerable. But maybe I am wrong.